core/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/core/model/HttpInput.kt

```diff
@@ -0,0 +1,141 @@
+/*
+ *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.alerting.core.model
+
+import org.apache.commons.validator.routines.UrlValidator
+import org.apache.http.client.utils.URIBuilder
+import org.elasticsearch.common.CheckedFunction
+import org.elasticsearch.common.ParseField
+import org.elasticsearch.common.Strings
+import org.elasticsearch.common.unit.TimeValue
+import org.elasticsearch.common.xcontent.NamedXContentRegistry
+import org.elasticsearch.common.xcontent.ToXContent
+import org.elasticsearch.common.xcontent.XContentBuilder
+import org.elasticsearch.common.xcontent.XContentParser
+import org.elasticsearch.common.xcontent.XContentParserUtils
+import java.io.IOException
+
+/**
+ * This is a data class of HTTP type of input for Monitors.
+ */
+data class HttpInput(
+    val scheme: String,
+    val host: String?,
+    val port: Int,
+    val path: String?,
+    val params: Map<String, String>,
+    val url: String,
+    val connection_timeout: Int,
```

Comment:
https://github.com/opendistro-for-elasticsearch/alerting/blob/38d2c20274cebb856e9cf20cda8c97b826fadbaf/core/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/core/model/HttpInput.kt#L41-L42

`connect_timeout` and `socket_timeout` should be part of the client settings not the input.

Reply:
These can be set from `HttpInput` for now, as XPack also allows to modify timeout values using inputs. Could probably make this available to set in `/_cluster/setting`

Reply:
Then we should bound these to an upper limit to avoid clients setting too large timeouts.

Reply:
Sure, thinking of using 60 seconds as the maximum since the highest frequency monitors can run is every 60s. Do you have any suggestions for the maximum number?

Reply:
`connect_timeout` should be 5 secs and `socket_timeout` can be 60 secs.

Reply:
`connection_timeout` changed to 5s

Reply:
Issue created regarding making the settings from `_cluster/settings` #86